src/applications/disability-benefits/all-claims/components/ComboBox.jsx

```diff
@@ -9,31 +29,216 @@ import React from 'react';
 export class ComboBox extends React.Component {
   constructor(props) {
     super(props);
+    // is there a cleaner way to pass this in?
+    this.disabilitiesArr = props.uiSchema['ui:options'].listItems;
     this.state = {
+      bump: false,
+      searchTerm: '',
       input: '',
+      value: '',
+      highlightedIndex: 0,
+      ariaLive1: '',
+      ariaLive2: '',
+      filteredOptions: [],
     };
+    this.inputRef = React.createRef();
+    this.listRef = React.createRef();
+  }
+
+  // Triggers updates to the list of items on state change
+  componentDidUpdate(prevProps, prevState) {
+    this.updateFilterOptions(prevState);
   }
 
   handleSearchChange = e => {
+    const { bump } = this.state;
     const newTextValue = e.target.value;
     this.setState({
+      searchTerm: newTextValue,
+      bump: !bump,
       input: newTextValue,
     });
     this.props.onChange(newTextValue);
   };
 
+  // update highlight class
+  handleMouseEnter(e, optionIndex) {
+    this.setState({ highlightedIndex: optionIndex });
+  }
+
+  // Filters list of conditions based on free-text input
+  filterOptions = () => {
+    const { searchTerm, value, bump } = this.state;
+    const options = this.disabilitiesArr;
+    const startsWith = options.filter(a =>
+      a.toLowerCase().startsWith(searchTerm.toLowerCase()),
+    );
```

Comment:
This is included in the fullStringSimilaritySearch function so it might be duplicated here.